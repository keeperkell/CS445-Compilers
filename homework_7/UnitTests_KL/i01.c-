## C-S22
int ga:111;
main() begin 
    output(ga);
     begin 
     end 
 end 
