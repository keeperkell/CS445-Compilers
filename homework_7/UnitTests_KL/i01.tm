* =================================================
* C- compiler version C-S22
* Built: Apr 17 - May 8
* Author: Keller Lawson
* File Compiled:  homework_7/UnitTests_KL/i01.tm
* =================================================
* 
* START FUNC INPUT
  1:     ST  3,-1(1)	Store return addr 
  2:     IN  2,2,2	Get int input 
  3:     LD  3,-1(1)	Load return addr 
  4:     LD  1,0(1)	Adjust fp 
  5:    JMP  7,0(3)	Return 
* END FUNC INPUT
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC OUTPUT
  6:     ST  3,-1(1)	Store return addr 
  7:     LD  3,-2(1)	Load return addr 
  8:    OUT  3,3,3	Output int 
  9:     LD  3,-1(1)	Load return addr 
 10:     LD  1,0(1)	Adjust fp 
 11:    JMP  7,0(3)	Return 
* END FUNC OUTPUT
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC INPUTB
 12:     ST  3,-1(1)	Store return addr 
 13:    INB  2,2,2	Get bool input 
 14:     LD  3,-1(1)	Load return addr 
 15:     LD  1,0(1)	Adjust fp 
 16:    JMP  7,0(3)	Return 
* END FUNC INPUTB
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC OUTPUTB
 17:     ST  3,-1(1)	Store return addr 
 18:     LD  3,-2(1)	Load return addr 
 19:   OUTB  3,3,3	Output bool 
 20:     LD  3,-1(1)	Load return addr 
 21:     LD  1,0(1)	Adjust fp 
 22:    JMP  7,0(3)	Return 
* END FUNC OUTPUTB
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC INPUTC
 23:     ST  3,-1(1)	Store return addr 
 24:    INC  2,2,2	Get char input 
 25:     LD  3,-1(1)	Load return addr 
 26:     LD  1,0(1)	Adjust fp 
 27:    JMP  7,0(3)	Return 
* END FUNC INPUTC
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC OUTPUTC
 28:     ST  3,-1(1)	Store return addr 
 29:     LD  3,-2(1)	Load return addr 
 30:   OUTC  3,3,3	Output char 
 31:     LD  3,-1(1)	Load return addr 
 32:     LD  1,0(1)	Adjust fp 
 33:    JMP  7,0(3)	Return 
* END FUNC OUTPUTC
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* START FUNC OUTNL
 34:     ST  3,-1(1)	Store return addr 
 35:  OUTNL  3,3,3	Output newline 
 36:     LD  3,-1(1)	Load return addr 
 37:     LD  1,0(1)	Adjust fp 
 38:    JMP  7,0(3)	Return 
* END FUNC OUTNL
* 
* ** ** ** ** ** ** ** ** ** ** ** **
* FuncK main
* LOFF Line96: -2
 39:     ST  3,-1(1)	Store return addr 
* COMPOUND
* LOFF Line188: -2
* START COMPOUND BODY
* START CALL output
 40:     ST  1,-2(1)	Store fp in ghost frame for  output
* START Param 1
* LOFF Line680: -3
* LOFF Line685: -4
* START ID
 41:     LD  3,0(0)	Load var ga
* END ID
 42:     ST  3,-4(1)	Push parameter 623 
* LOFF Line615: -5
* END Param 1
 43:    LDA  1,-2(1)	Ghost frame becomes new active frame 
 44:    LDA  3,1(7)	Load return addr 
* BackPatch 45
 45:    JMP  7,-40(7)	CALL OUTPUT output
 46:    LDA  3,0(2)	Save the result in ac 
* END CALL
* LOFF Line749: -2
* COMPOUND
* LOFF Line188: -2
* START COMPOUND BODY
* END COMPOUND BODY
* END COMPOUND
* END COMPOUND BODY
* END COMPOUND
 47:    LDC  2,0(6)	Set return value to 0 
 48:     LD  3,-1(1)	Load return address 
 49:     LD  1,0(1)	Adjust fp 
 50:    JMP  7,0(3)	return 
* End of Function main
* 
* =========================================================
  0:    JMP  7,50(7)	Jump to init 
* START INIT
 51:    LDA  1,-1(0)	Set first frame after globals 
 52:     ST  1,0(1)	Store old fp 
* START INIT GLOBALS AND STATICS
 53:    LDC  3,111(6)	Load int const 
 54:     ST  3,0(0)	Store Var ga
* END INIT GLOBALS AND STATICS
 55:    LDA  3,1(7)	Load return address 
 56:    JMP  7,-18(7)	Jump to main 
 57:   HALT  0,0,0	DONE 
* END INIT
