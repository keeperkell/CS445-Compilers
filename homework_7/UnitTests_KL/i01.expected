Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: UnitTests/i01.tm
111
Bye.
